### ISSUE TITLE
return new Promise arrow function, await not being called asynchronously

### ISSUE THREAD

**Issue by Gerrudo** [Gerrudo/RESTaurants-Web-API#9]
```
        return new Promise (async (resolve) => {
            //The 2 await result declarations are not running async
            if (isCached === true){
                let result = await this.getResultsFromCache(randomPlace);
                resolve(result);
            }else{
                let result = await this.newRequest(randomPlace);
                resolve(result);
            };
        });
```

The above block works in theory, but the 2 await calls are not being run asynchronously, and returning undefined to res.

**Comment by Gerrudo** [comment]
Found this when looking with the linter:
![image](https://user-images.githubusercontent.com/43029469/100545263-e07e1b00-3252-11eb-85f0-da19e7b1e7d5.png)
This might be the start of our problem, we may just be returning undef from the promise call, looking at the promises that's being used in the method being called, the scope the the return New Promise may be being misused.

**Comment by Gerrudo** [comment]
Issue resolved. Problem was return new Promise scoping was incorrect:

Incorrect, returns undefined:
![image](https://user-images.githubusercontent.com/43029469/100545522-25568180-3254-11eb-8f5a-95cb226772a5.png)

Correct, await now receives our promise:
![image](https://user-images.githubusercontent.com/43029469/100545553-47e89a80-3254-11eb-8f93-8dd1e765aac2.png)

The problem being, our return statement was scope down to inside another function, so the promise was being returned to that method alone, and not to higher up, where we called the method. 
The correct screenshot shows our return new Promise statement returning at the top of the function, which resolves the promise back to where we called it; from our other method. 

Side note:
It's always best to have our return new Promise statements at the top of our methods, reason being is that when the method is called, it always returns to the right place, and that if we need to return reject at any point incase of an error or bad data, we can return it anywhere in the method.
